qiskit/providers/ibmq/job/queueinfo.py

```diff
@@ -56,5 +62,47 @@ def __init__(
         self.hub_priority = hub_priority
         self.group_priority = group_priority
         self.project_priority = project_priority
+        self.job_id = job_id
 
         super().__init__(**kwargs)
+
+    def __repr__(self) -> str:
+        """"""
+        queue_info = ["{}={}".format(attr, repr(value))
+                      for attr, value in self.__dict__.items()]
+
+        return "{}({})".format(self.__class__.__name__, ', '.join(queue_info))
+
+    def format(self) -> str:
+        """Build an user-friendly report for the job queue information.
+
+        Returns:
+             The job queue information report.
+        """
+        status = api_status_to_job_status(ApiJobStatus(self._status)).value \
+            if self._status else self._get_value(self._status)
+        estimated_start_time = arrow.get(self.estimated_start_time).humanize() \
+            if self.estimated_start_time else self._get_value(self.estimated_start_time)
+        estimated_complete_time = arrow.get(self.estimated_complete_time).humanize() \
+            if self.estimated_complete_time else self._get_value(self.estimated_complete_time)
+
+        queue_info = [
+            "Job {} queue information:".format(self._get_value(self.job_id)),
+            "    queue position: {}".format(self._get_value(self.position)),
+            "    status: {}".format(status),
+            "    estimated start time: {}".format(estimated_start_time),
+            "    estimated completion time: {}".format(estimated_complete_time),
+            "    hub priority: {}".format(self._get_value(self.hub_priority)),
+            "    group priority: {}".format(self._get_value(self.group_priority)),
+            "    project priority: {}".format(self._get_value(self.project_priority))
+        ]
+
+        return '\n'.join(queue_info)
+
+    def _get_value(self, value: Optional[Any], _default_value: str = 'unknown') -> Optional[Any]:
+        """Returns the value if it exists or a default.
+
+        Returns:
+            The value if it is not None, else a default value.
+        """
+        return value if value is not None else _default_value
```

Comment:
nit: you can shorten this to `return value or _default_value`. And why is `_default_value` private?

Reply:
Thanks for the tip @jyu00! I was just trying to convey that it's not really something to change, but I don't think it matters if it's changed by the user...